src/Acrobot.jl

```diff
@@ -0,0 +1,214 @@
+import Flux.Tracker
+
+mutable struct AcrobotEnv
+    viewer
+    observation_space
+    action_space
+    state
+    dt
+    LINK_LENGTH_1  # [m]
+    LINK_LENGTH_2  # [m]
+    LINK_MASS_1  # [kg] mass of link 1
+    LINK_MASS_2  # [kg] mass of link 2
+    LINK_COM_POS_1  # [m] position of the center of mass of link 1
+    LINK_COM_POS_2  # [m] position of the center of mass of link 2
+    LINK_MOI  # moments of intertia for both links
+    MAX_VEL_1
+    MAX_VEL_2
+    AVAIL_TORQUE
+    torque_noise_max
+    book_or_nips
+end
+
+
+function AcrobotEnv()
+    dt = 2f-1
+    LINK_LENGTH_1 = 1f0
+    LINK_LENGTH_2 = 1f0
+    LINK_MASS_1 = 1f0
+    LINK_MASS_2 = 1f0
+    LINK_COM_POS_1 = 5f-1
+    LINK_COM_POS_2 = 5f-1
+    LINK_MOI = 1f0
+
+    MAX_VEL_1 = Float32(4 * pi)
+    MAX_VEL_2 = Float32(9 * pi)
+
+    AVAIL_TORQUE = [-1f0, 0f0, 1f0]
+    torque_noise_max = 0f0
+
+    # use dynamics equations from the nips paper or the book
+    book_or_nips = "book"
+    # action_arrow = nothing
+    # domain_fig = nothing
+    # actions_num = 3
+
+    viewer = nothing
+    high = [1f0, 1f0, 1f0, 1f0, MAX_VEL_1, MAX_VEL_2] #//
+    low = -high
+    observation_space = (low, high)
+    action_space = 1:3
+    state = nothing
+
+    AcrobotEnv(viewer, observation_space, action_space, state, dt, LINK_LENGTH_1,
+               LINK_LENGTH_2, LINK_MASS_1, LINK_MASS_2, LINK_COM_POS_1, LINK_COM_POS_2,
+               LINK_MOI, MAX_VEL_1, MAX_VEL_2, AVAIL_TORQUE, torque_noise_max, book_or_nips)
+end
+
+
+function reset!(env::AcrobotEnv)
+    env.state = param(rand(Float32, 4) * 2f-1 .- 1f-1)
+    return _get_ob(env)
+end
+
+#=
+function custom_reset!(env::AcrobotEnv, args=[-0.04616826, -0.04653378,  0.07321809, -0.04461966])
+    env.state = args
+    return _get_ob(env)
+end
+=#
+
+function step!(env::AcrobotEnv, action)
+    s = env.state
+    torque = env.AVAIL_TORQUE[action]
+
+    # Add noise to the force action
+    env.torque_noise_max > 0 && (torque += param(env.torque_noise_max * (2rand(Float32) - 1)))
+
+    # Now, augment the state with our force action so it can be passed to _dsdt
+    s_augmented = vcat(s, torque)
+
+
+    ns = rk4(env, _dsdt, s_augmented, [0, env.dt])
+    # only care about final timestep of integration returned by integrator
+    ns = ns[end, :]
+    ns = ns[1:4]  # omit action
+    ns[1] = wrap(ns[1], -pi, pi)
+    ns[2] = wrap(ns[2], -pi, pi)
+    ns[3] = bound(ns[3], -env.MAX_VEL_1, env.MAX_VEL_2)
+    ns[4] = bound(ns[4], -env.MAX_VEL_1, env.MAX_VEL_2)
+    env.state = ns
+    terminal = _terminal(env)
+    reward = !terminal ? -1f0 : 0f0
+
+    return (_get_ob(env), reward, terminal, Dict())
+end
+
+
+function _get_ob(env::AcrobotEnv)
```

Comment:
Follow [this](https://github.com/FluxML/Gym.jl/blob/cd02fa23bc794723fec1abb632af4bc0dd4bb0c2/src/Pendulum.jl#L54) pattern in `_get_obs` in order to make it differentiable.

Reply:
Will this have any effect on the processing time? I mean, dealing with scalars is easier than dealing with an array of size one...

Reply:
Right now Flux does not offer the same features with TrackedReal as it does with TrackedArray, hence we treat scalar which are to be tracked as array